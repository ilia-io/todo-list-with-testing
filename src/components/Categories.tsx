import React from 'react';

const categories = ['Все', 'Активные', 'Выполненные'];

interface ICategoryProps {
  categoryState: number;
  setCategoryState: (id: number) => void;
  taskLeft: number;
  clearCompleted: () => void;
}

const Categories: React.FC<ICategoryProps> = ({
  categoryState,
  setCategoryState,
  taskLeft,
  clearCompleted,
}) => {
  return (
    <div className="actions">
      <p data-testid="text-test">
        Осталось сделать: <span>{`${taskLeft}`}</span>
      </p>
      {categories.map((category, index) => (
        <button
          data-testid="categ-btn"
          onClick={() => setCategoryState(index)}
          className={categoryState === index ? 'active' : ''}
          type="button"
          key={index}
        >
          {category}
        </button>
      ))}
      <button onClick={clearCompleted} type="button">
        Очистить выполненные
      </button>
    </div>
  );
};

export default Categories;
