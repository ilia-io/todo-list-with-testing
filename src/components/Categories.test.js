import React from 'react';
import { render, screen } from '@testing-library/react';
import Categories from './Categories';

describe('Categories component tests', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    render(<Categories />, container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container.remove();
  });

  it('renders correctly initial document', () => {
    const buttons = screen.getAllByRole('button');
    expect(buttons).toHaveLength(4);
    expect(buttons).toMatchSnapshot();
  });
});

describe('Categories TEST', () => {

  it('Buttons renders', () => {
     render(<Categories />);
    const buttons = screen.getAllByRole('button');
    expect(buttons).toHaveLength(4);
    expect(buttons).toMatchSnapshot();
  });

  it('text renders', () => {
    render(<Categories />);
    const p = screen.getByTestId('text-test');
    expect(p).toBeInTheDocument();
  });
});
