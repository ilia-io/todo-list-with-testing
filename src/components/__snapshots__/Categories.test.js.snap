// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component tests renders correctly initial document 1`] = `
Array [
  <button
    class=""
    type="button"
  >
    Все
  </button>,
  <button
    class=""
    type="button"
  >
    Активные
  </button>,
  <button
    class=""
    type="button"
  >
    Выполненные
  </button>,
  <button
    type="button"
  >
    Очистить выполненные
  </button>,
]
`;
